presto-thrift/src/main/java/io/prestosql/plugin/thrift/ThriftMetadata.java

```diff
@@ -183,6 +164,64 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
         }
     }
 
+    @Override
+    public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
+    {
+        ThriftTableHandle handle = (ThriftTableHandle) table;
+        return new ConnectorTableProperties(handle.getConstraint(), Optional.empty(), Optional.empty(), Optional.empty(), ImmutableList.of());
```

Comment:
I think this should use `new ConnectorTableProperties()` so that the constraint is `TupleDomain.all()`, as the table constraint is just a hint to the backend and is not guaranteed. The original code returned the passed in constraint as the unenforced constraint, which means no guarantees.

Reply:
Fixed it.